### PR TITLE
[#D3D-2461] Updates tinygltf

### PR DESCRIPTION
**Pull request by AurL**

### CODE REVIEW

File: tiny_gltf_loader.h [hunk 1]
```diff
@@ -140,9 +140,21 @@ namespace tinygltf {
 #define TINYGLTF_SHADER_TYPE_VERTEX_SHADER (35633)
 #define TINYGLTF_SHADER_TYPE_FRAGMENT_SHADER (35632)
 
+// This is an unofficial flag that allows to know if the flag needs to be
+// enabled or not for this texture. glTF spec considers it as false by
+// default but most of the renderers enable it.
+// Unknown will be considered the same as false, and means that the flag is not given
+// in the file
```
Comment (marchelbling) [inline tiny_gltf_loader.h:147]: 👍

File: tiny_gltf_loader.h [hunk 2]
```diff
@@ -204,6 +224,23 @@ typedef struct {
   ParameterMap values;
 } Material;
 
+//FIXME: the spec is not consistant with the official samples for this king of materials
+// so this function will need some updates
+typedef struct {
+  std::string name;
+  bool doubleSided;
+  int jointCount;
```
Comment (marchelbling) [inline tiny_gltf_loader.h]: really useful? and is it really a material property?

File: tiny_gltf_loader.h [hunk 3]
```diff
@@ -897,6 +940,41 @@ static bool ParseNumberProperty(double *ret, std::string *err,
   return true;
 }
 
+static bool ParseJSONProperty(std::map<std::string, std::string> *ret, std::string *err,
```
Comment (marchelbling) [inline tiny_gltf_loader.h:943]: I don't get why people would use a pointer and not a reference (I know it's the tinygltf style but still)

File: tiny_gltf_loader.h [hunk 4]
```diff
@@ -897,6 +940,41 @@ static bool ParseNumberProperty(double *ret, std::string *err,
...
+        (*err) += "'" + property + "' property is missing. \n'";
+      }
+    }
+    return false;
+  }
+
+  if(!it->second.is<picojson::object>()) {
+    if (required) {
+      if (err) {
+        (*err) += "'" + property + "' property is not an array.\n";
```
Comment (marchelbling) [inline tiny_gltf_loader.h]: error message looks wrong; you're testing `is<picojson::object>` not `is<picojson::array>`

File: tiny_gltf_loader.h [hunk 5]
```diff
@@ -1213,11 +1291,16 @@ static bool ParseImage(Image *image, std::string *err,
...
+      // Keep texture path (for textures that cannot be decoded)
+      image->path = uri;
+
+      // If the image cannot be loaded, keep uri.
       if (!LoadExternalFile(&img, err, uri, basedir, 0, false)) {
         if (err) {
           (*err) += "Failed to load external 'uri'. for image parameter\n";
         }
-        return false;
+        return true;
```
Comment (marchelbling) [inline tiny_gltf_loader.h]: that's a bold change; if it's a bug, please make a separate commit

File: tiny_gltf_loader.h (hunk 2 above)
Comment (AurL) [reply tiny_gltf_loader.h]: Useless, I remove it

File: tiny_gltf_loader.h (hunk 4 above)
Comment (AurL) [reply tiny_gltf_loader.h]: Nice catch

**Review by marchelbling — APPROVED** [review]
